File: codejam/2021/Round1A/a.py

t = int(input())
def inc(i,l):
    if l[i+1] > l[i]:
        return 0
    nex = str(l[i+1])
    pre = str(l[i])
    if pre[:len(nex)] == nex and not all(map(lambda x:x=="9",pre[len(nex):])):
        l[i+1] = l[i]+1
        return len(pre)-len(nex)
    ret = 0
    #print(i,l[i],l[i+1])
    while True:
        for j in range(10):
            if l[i] < 10*l[i+1]+j:
                l[i+1] = 10*l[i+1]+j
                return ret+1
        l[i+1] = 10*l[i+1]
        ret+=1

def solve_one(pn):
    n = int(input())
    l = list(map(int,input().split()))
    ret = 0
    for i in range(n-1):
        ret += inc(i,l)
    print(f"Case #{pn}: {ret}")

for i in range(t):
    solve_one(i+1) 

File: codejam/2021/Round1A/a.cpp
#include "bits/stdc++.h"
using namespace std;

#define FOR(i,a,b) for(int i=(a);i<(b);++i)
#define REP(i,n)   FOR(i,0,n)
#define ALL(a)  (a).begin(),(a).end()
#define LL long long

int t,n;
LL x[100];

void solve_one(int problem_num){
    cin>>n;
    REP(i,n)cin>>x[i];
    int ret=0;

    cout<<"Case #"<<problem_num<<": "<<ret<<endl;
}

int main(){
  cin>>t;
  REP(i,t)solve_one(i+1);
  return 0;
}
